### DISCUSSION TITLE
Hide unnecessary app volume [Feature]

### DISCUSSION THREAD

**Discussion by CosimoMatassini** [File-New-Project/EarTrumpet#713] [Ideas]
It could be useful to be able to hide an app from EarTrumpet just because I don't use it (like "System Sounds") in order not to cover too much screen and be focused on what I work with.

**Comment by Drexl** [comment]
Yes, please. Also, just generally having each audio source take up less horizontal space on the mixer would be very useful. I often have the mixer open and extended to display every audio source and it takes up a very large amount of real-estate. I'd like to see the sliders shrunk by half, and the ability to hide the system name for each device (just show the user-defined name). This would make the mixer a lot more useable for me.

**Comment by celluj34** [comment]
Adding on to this, maybe even the ability to "pin" certain apps so they show above others.

**Reply by ghost** [reply]
open a request for this please

**Comment by ryantheleach** [comment]
+1 to pinning apps/devices and hiding them.

In case this is an XYproblem I figured I should explain why.

![image](https://user-images.githubusercontent.com/141554/100442743-75203600-30f8-11eb-89aa-2f6adb9ee296.png)

System Sounds and Discord are currently listed for EVERY device, no matter if they are making noise or not, or whether they are being played to a specific sound device configured inside the program.

Generally speaking, I don't ever need to adjust these on devices they are not outputting on, so they just add to the clutter.

Ideally System Sounds and Discord would only show up on the devices they are actually using. But I'm guessing this is a 3rd party bug due to keeping a sound session open for some reason or another.

**Reply by YoDawgYo** [reply]
I'd love to have the ability to pin certain devices to 100% volume so the external volume knob can be used instead (like on my SCHIIT headphone amp) and leave other devices set to variable (like powered usb speakers) so my usb/bluetooth volume knob can control that (via windows volume).

**Reply by Veri7ion** [reply]
@YoDawgYo BetterTrumpet can do this its a fork of EarTrumpet

**Comment by c-dot-gonz** [comment]
I have a lot of audio inputs/outputs that are needed frequently enough that I can't just disable them, but I never actually run programs to them (they're hardware inputs that I need for recording) so being able to just hide them from EarTrumpet would be amazing.

**Comment by LukasDoubleU** [comment]
I implemented this and other features over at my fork.
You can right click an app and then 'delete' or 'hide' it.
There is a button in the settings to restore all hidden apps.

If you want you can try out the pre-release:
https://github.com/LukasDoubleU/EarTrumpet/releases

![image](https://user-images.githubusercontent.com/16664756/106393144-b1705b00-63f5-11eb-84ee-817515512061.png)

**Reply by Coldblackice** [reply]
This is awesome. Any chance of merging it with the latest/development release?

**Reply by mrchanshares** [reply]
The forked source code does not match with the pre-release yet. Why?

**Reply by nnyj** [reply]
I don't know how to do a pull request, so I just forked and ported the hideApps and changeComms features to the latest development release.

The repo if anyone wants to try: https://github.com/nnyj/EarTrumpet

**Reply by Drexl** [reply]
The problem with this, for me, is that it hides the application from all audio devices, not just one. Also, even if all applications are hidden from a device that device still continues to show. Even with all offending applications hidden I have so many devices I have to scroll to find what I want, and I do scroll which of course means I'm constantly changing volume levels because I forget to pixel hunt for the scroll bar on the right rather than using my mouse-wheel.

This is a great start, but it doesn't actually solve the problem I'm having unfortunately. :(

**Reply by Mlkjhgfd** [reply]
Is it possible to have an updated fork please ?

**Comment by dannywrayuk** [comment]
Such a spooky coincidence! I just sat down an hour ago and began implementing this feature because I really needed it. This saves me a load of effort! Are you planning on opening a PR?

**Comment by LukasDoubleU** [comment]
Yes, I wrote a mail to the devs and am waiting for their response.
If I get their OK I will open a PR

**Comment by celluj34** [comment]
You don't need "approval" to make a PR, just do it...

**Reply by mrchanshares** [reply]
In such case, is it needed to change the exe name? or also the project/product name?

**Comment by davuxcom** [comment]
We kindly ask that in order to ensure your effort isn't wasted, we discuss PRs before submitting them. We still review and consider all PRs however if we collaborate early it's more likely that the technical solution will match the project goals.

https://github.com/File-New-Project/EarTrumpet/blob/master/CONTRIBUTING.md

**Comment by ghost** [comment]
It has been a year now, does anyone know if that PR ever got opened? Asking because the delete feature seems really nice to have :p

**Reply by LukasDoubleU** [reply]
I did make a PR back then. But it didn't get merged because of some stuff that was being developed in the background.

**Reply by mrchanshares** [reply]
> I did make a PR back then. But it didn't get merged because of some stuff that was being developed in the background.

Try to submit your changes to your fork. And if possible, can you also help me in some code change? Such as make the mixer window transparent and also add 2 volume bars below the current one for left and right channels.

**Comment by Coldblackice** [comment]
I was just about to make these same suggestions, A.) ability to condense entries (app volumes and/or sound devices), and/or B.) ability to hide specific devices/app volumes.

**Reply by bthames** [reply]
I have a lot of junk devices from the Steam Streaming service and scrolling through tons of inputs I never use really drags my user experience down.

**Reply by riverar** [reply]
@bthames Have you tried disabling them in Windows?

**Reply by DaEkstrim** [reply]
> @bthames Have you tried disabling them in Windows?

Disabling the devices is not a good option, as while I don't want to assign software to them, they're still needed by certain applications. In cases such as Voiceemeter, disabling the device would not allow Voiceemeter to use them. So while disabling devices hides them from EarTrumpet, it's has major adverse effects. It's like dropping napalm on the house to kill the spider.

**Reply by DaEkstrim** [reply]
I tried a rudimentary approach (I have a solid backend development background, but very little XAML experience), using a text box at the bottom of the list to filter devices.

![image](https://user-images.githubusercontent.com/18722053/172050350-1918db56-570e-4bc7-99e3-d676b2ecb571.png)
![image](https://user-images.githubusercontent.com/18722053/172050456-3ac17a35-7da9-4e28-89da-bdbd7c2bc255.png)

My idea was to allow users to filter devices based on the value in the text box, (using comma to specify multiple names) to filter on the DisplayName of the devices they want to see. This value is saved in the AppSettings so that it's reloaded and applied on app start. The changes are applied on key press leading to some janky user experience, so I think switching that to 'OnEnter' might be better or maybe perhaps a delayed effect (but that's way above my head on how it would be implemented in the XAML binding events system).

I did this in a duct-tape way, so  it's not PR worthy yet. But if there's interest, I can improve and document the changes a bit.

**Reply by ryantheleach** [reply]
Whilst a search that persists is interesting, it doesn't align with the solution I had envisioned when subscribing to this thread, and feels like a delicate solution.

I envisaged a solution would have a menu for hiding apps, maybe hidden in a right click menu, that would make them no longer appear in the list aside from an icon indicating that a device had active hidden applications assigned.

A hidden app would only hide the ENTRY, and not all instances of the application. Yes this would mean manually hiding each entry, but it provides flexibility needed when the app is likely to output on *one* of the many devices it has active sessions open for. This is different to Lukes fork.

![image](https://user-images.githubusercontent.com/141554/172073195-5aae65b4-1d7c-4fc4-85ce-19ca08227a06.png)

In my instance, I would hide System Sounds, Discord, Steam, for nearly every device.

App Entries with non-default volumes/mute status should not be hidden, to be transparent to the user.

Devices could also be suppressed in a similar manner, but note that most people in the issue are requesting that App Entries can be hidden, not all instances of Apps, or Devices, Although I can see the utility of hiding Devices.

A poor mockup made in paint: 

![image](https://user-images.githubusercontent.com/141554/172073439-e5c772c7-8217-4691-8d15-e4ffbafd6c37.png)

**Comment by Drexl** [comment]
I've been dealing with having to scroll twice the vertical resolution of my display for a long time now. I hope someone manages to let us hide these sources eventually. I also can't disable them. They are constantly in use, but there is never a situation where I want to adjust them via EarTrumpet

**Reply by fishwithfeet** [reply]
I feel you

**Reply by ryantheleach** [reply]
Ultimately this is the root cause of wanting this feature, aside from clutter.

It's almost entirely caused by Apps maintaining sessions on all devices they have no right to.

Steam, System Sounds and Discord are the primary offenders for me.

![image](https://user-images.githubusercontent.com/141554/172073483-f8a14540-4126-44f9-b706-073eee320777.png)

**Reply by Drexl** [reply]
Same, but there are also entire devices I don't need to see.

**Reply by Coldblackice** [reply]
> Ultimately this is the root cause of wanting this feature, aside from clutter.
> 
> It's almost entirely caused by Apps maintaining sessions on all devices they have no right to.
> 
> Steam, System Sounds and Discord are the primary offenders for me.
@ryantheleach 

Is there a way to block apps from doing this? Super frustrating.

**Comment by lyndonguitar** [comment]
Google brought me here, it would be nice to have this feature, as I cannot disable all the sound systems as sometimes they are used by apps (like Steam Remoteplay, Moonlight, parsec, etc)

**Comment by ghost** [comment]
Also, very important is that the hidden sound devices should not be controllable by using the volume up/down keys on your keyboard (or laptop, or whatever). I personally don't really mind having a slightly longer list of audio devices, however EarTrumpet likes to switch the selected device everytime something changes and sets it to a virtual one (I'm using ASIOLinkPro so I can have multiple audio sources all go through my ASIO driver), which does not let me actually change my system volume as I'd like to. This means I constantly have to right-click the EarTrumpet tray icon and change the selection everytime I open/close my DAW in order to actually change my system volume using my keyboard.

**Comment by IFD3** [comment]
I have the same problem but I really want to hide some devices aswell. I really don't need steam streaming devices volume controlled and several others too. right now I just deactivate them using a script when I don't need them.
Mainly because I have a couple of real devices in parallel and don't want to scroll 3 screens down because the app doesn't even properly remembers the scroll position

**Comment by ghost** [comment]
auto hide applications which is not playing any audio/sound

**Reply by ryantheleach** [reply]
I can't get behind this change.

**Reply by ghost** [reply]
> auto hide applications which is not playing any audio/sound

This would be an anti-feature. This way volume would not be adjustable for apps that play audio on demand only (such as music software or for MIDI instruments) and audio cannot be turned down before playing something you know is very loud for example.

**Comment by Parsnip** [comment]
+1 from me as well. I love ear trumpet but I would love to customize what's shown in the systray menu. Hide everything but the master volume, hide specific devices and hide specific sound sources would be the 3 ideal settings for me personally.

**Comment by Runkel79** [comment]
Would love to see a function like this.
I'm surprised this function isn't already build in because everything else is so well thought out and executed :)

**Comment by rmack00** [comment]
Another screenshot for the pile... Unfortunately this negates any convenience the app would provide 

![Screenshot 2024-04-15 105226](https://github.com/File-New-Project/EarTrumpet/assets/167114064/13130162-7f50-4eef-93ee-5a26e1f801b9)

**Comment by charmarkk** [comment]
Another one coming from Google, looking for a way to do this. 
![image](https://github.com/File-New-Project/EarTrumpet/assets/22264269/f89ba627-e927-4304-80ca-dc9605f9ac48)

**Comment by geextahslex** [comment]
Is there any progress on this feature? 4 years have passed...

**Reply by charmarkk** [reply]
> Try clicking the arrow top right corner

As many have noted elsewhere in the ticket, that only collapses everything that's not the default output. The ticket requests the ability to remove _totally_ unused (or undesired) outputs from EarTrumpet without disabling them. Some may need virtual audio outputs enabled for XYZ reason, but don't want them to show up as a device in EarTrumpet (again, as many have stated in the ticket). E.g., my screenshot, I don't need Steam and Oculus virtual outputs visible, but I can't disable them as they're needed for other services.

**Comment by qqqianye** [comment]
Another one coming from Google , also waiting for new versions

**Comment by FuglyShmugly** [comment]
You can rename outputs like this to keep them on top, while we wait for the hide feature :)
![image](https://github.com/user-attachments/assets/f9b2ca5c-e223-436d-aa7f-046272bd7242)

**Reply by Coldblackice** [reply]
Clever idea! Unfortunately, this doesn't alter my Ear Trumpet's flyout panel. Does it change your flyout's ordering?

I don't know how ET is ordering my flyout, honestly, as it's not alphabetical nor date-added/modified. Frustrating.

**Comment by Veri7ion** [comment]
This has been requested for years

**Comment by snoopsneep** [comment]
Throwing a bump in the ring. The last great feature EarTrumpet needs, and I didn't know we needed it until Flow Launcher and ShareX decided they wanna sit in the EarTrumpet menu 24/7. Let us reduce clutter!!!

**Comment by Esashiero** [comment]
Bumping this, with voice meeter and other useless output device, it's almost annoying to use. Hopefully the devs will hear us at some point... it's not even that crazy of a change, I don't undersand why it's not been added before.

**Comment by momentmal** [comment]
+bump, totally would be useful, this is the reason i was looking for s.th. like eartrumpet in the first place. i have the same disadvantage of a long device list in the stock windows device selector already.

**Comment by alphaqu** [comment]
Hey! For anyone wanting this feature, I have updated a fork (https://github.com/nnyj/EarTrumpet) with this feature to the latest version of EarTrumpet, and published a [release here](https://github.com/alphaqu/EarTrumpet/releases/tag/2.3.0.0%2Bnnyj). 

Have a nice day!

**Reply by Veri7ion** [reply]
Thank you for making this it works great for fully hiding specific programs, however I think many of us were also looking for the ability to hide duplicate programs under multiple audio devices. For example the screenshot above from @charmarkk posted on May 25, 2024. Or in the screenshot below, all the highlighted program sliders all duplicates that I'd love to be able to hide. Unfortunately your fork hides the programs in **all** audio devices. Would it be possible to add the feature to hide programs for specific audio devices only? That would be amazing 😅
![25-04-2025 EarTrumpet_1314](https://github.com/user-attachments/assets/daebc033-de2d-4170-ba14-d39d7b82c06e)

**Reply by charmarkk** [reply]
That or hide unnecessary devices/lists _altogether_ (my preference). In my screenshot, I can't disable many of those, as they're needed for certain occasional instances, but I'd like to be able to hide the devices I don't need (like the Oculus speakers, or Steam Streaming Speakers). That would solve everything being asked about in this issue, I think.

**Comment by yotengodormir** [comment]
Would love this feature. My keyboard driver adds a volume control to the fly out list - really dont need it there.

**Reply by Veri7ion** [reply]
take a look at https://github.com/xammen/BetterTrumpet 😉
